sunkit_image/offlimb_enhance.py

```diff
@@ -296,3 +298,158 @@ def normalizing_radial_gradient_filter(
+    may have to be changed in order to obtain a good-looking plot.
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        A SunPy map
+
+    radial_bin_edges : `~astropy.units.Quantity`
+        A two-dimensional array of bin edges of size [2, nbins] where nbins is
+        the number of bins.
+
+    order : `~int`
+        Order (Number) of fourier coefficients.
+
+    ratio_mix : `~float`
+        The ratio in which the original image and filtered image are mixed.
+
+    scale : None | `astropy.units.Quantity`
+        The radius of the Sun expressed in map units.  For example, in typical
+        helioprojective Cartesian maps the solar radius is expressed in units
+        of arcseconds.  If None, then the map scale is used.
+
+    intensity_summary :`~function`
+        A function that returns a summary statistic of the radial intensity,
+        for example `~numpy.mean` and `~numpy.median`.
+
+    intensity_summary_kwargs : None | `~dict`
+        Keywords applicable to the summary function.
+
+    width_function : `~function`
+        A function that returns a summary statistic of the distribution of intensity,
+        at a given radius, for example `~numpy.std`.
+
+    width_function_kwargs : `~function`
+        Keywords applicable to the width function.
+
+    application_radius : `~astropy.units.Quantity`
+        The FNRGF is applied to emission at radii above the application_radius.
+
+    number_angular_segments : `~int`
+        Number of angular segments in a circular annulus. Default value is 130. It should always be
+        even.
+
+    Returns
+    -------
+    new_map : `sunpy.map.Map`
+        A SunPy map that has had the FNRGF applied to it.
+    """
+
+    # Get the radii for every pixel
+    map_r = find_pixel_radii(smap).to(u.R_sun)
+
+    # Get the Helioprojective coordinates of each pixel
+    x, y = np.meshgrid(*[np.arange(v.value) for v in smap.dimensions]) * u.pix
+    coords = smap.pixel_to_world(x, y).transform_to(frames.Helioprojective)
+
+    # Get angles associated with every pixel
+    angles = np.arctan2(coords.Ty.value, coords.Tx.value)
+
+    # Making sure all angles are between (0, 2 * pi)
+    angles = np.where(angles < 0, angles + (2*np.pi), angles)
+
+    # Number of radial bins
+    nbins = radial_bin_edges.shape[1]
+
+    # Storage for the filtered data
+    data = np.zeros_like(smap.data)
+
+    # Iterate over each circular ring
+    for i in range(0, nbins):
+
+        # Finding the pixels which belong to a certain circular ring
+        annulus = np.logical_and(map_r > radial_bin_edges[0, i], map_r < radial_bin_edges[1, i])
+        annulus = np.logical_and(annulus, map_r > application_radius)
+
+        # The angle subtended by each segment
+        segment_angle = 2*np.pi / number_angular_segments
+
+        # Storage of mean and standard deviation of each segment in a circular ring
+        E = np.zeros((1, number_angular_segments))
```

Comment:
The variable name "E" is not descriptive.  Please give it a different name.

Reply:
Done.